### PR TITLE
Nutanix cusomter resource

### PR DESCRIPTION
**Pull request by sgaoshang**

### CODE REVIEW

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.16-nutanix-ipi-zones-single-os-image-customized-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.16-nutanix-ipi-zones-single-os-image-customized-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.16-nutanix-ipi-zones-single-os-image-customized-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.16-nutanix-ipi-zones-single-os-image-customized-f28

**Comment by sgaoshang** [comment]
@jianlinliu pls help to review it, thanks.

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.16.yaml [hunk 1]
```diff
@@ -297,7 +297,7 @@ tests:
     test:
     - ref: cucushift-installer-check-cluster-health
     workflow: cucushift-installer-rehearse-gcp-upi-xpn
-- as: nutanix-ipi-zones-single-os-image-f28
+- as: nutanix-ipi-zones-single-os-image-customized-f28
```
Comment (jianlinliu) [inline ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.16.yaml:300]:
```suggestion
- as: nutanix-ipi-singlezone-osimage-customresource-f28
```

File: ci-operator/step-registry/cucushift/installer/rehearse/nutanix/ipi/zones/provision/cucushift-installer-rehearse-nutanix-ipi-zones-provision-chain.yaml [hunk 2]
```diff
@@ -3,12 +3,9 @@ chain:
   steps:
   - chain: ipi-conf-nutanix
   - ref: ipi-conf-nutanix-zones
-  - ref: ipi-conf-osimage-nutanix
   - chain: ipi-install
   - ref: configure-registry-storage-nutanix
   - ref: enable-qe-catalogsource
   - chain: cucushift-installer-check
-  - ref: cucushift-installer-check-nutanix-zones
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/nutanix/ipi/zones/provision/cucushift-installer-rehearse-nutanix-ipi-zones-provision-chain.yaml:11]: I do not think we should remove this step from the zone chain.

File: ci-operator/step-registry/cucushift/installer/check/nutanix/customized-resource/cucushift-installer-check-nutanix-customized-resource-commands.sh [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+check_result=0
+
+function check_cpus() {
+   node=$1 cpus=$2
+   vcpuSockets=$(oc get machine -n openshift-machine-api "$node" -o jsonpath='{.spec.providerSpec.value.vcpuSockets}')
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/nutanix/customized-resource/cucushift-installer-check-nutanix-customized-resource-commands.sh:11]: Eventually the setting from install-config.yaml will be passed down to machine itself, machine-api yaml file is just an intermediate step, can we run nutanix client command line to check the setting from the vm instance?

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.16-nutanix-ipi-zones-single-osimage-customized-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.16-nutanix-ipi-zones-single-osimage-customized-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.16-nutanix-ipi-zones-single-osimage-customized-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.16-nutanix-ipi-zones-single-osimage-customized-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.16-nutanix-ipi-zones-single-osimage-customized-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.16-nutanix-ipi-zones-single-osimage-customized-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.16-nutanix-ipi-zones-single-osimage-customized-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.16-nutanix-ipi-zones-single-osimage-customized-f28

File: ci-operator/step-registry/cucushift/installer/check/nutanix/customized-resource/cucushift-installer-check-nutanix-customized-resource-commands.sh (hunk 3 above)
Comment (sgaoshang) [reply ci-operator/step-registry/cucushift/installer/check/nutanix/customized-resource/cucushift-installer-check-nutanix-customized-resource-commands.sh]: fixed, changed to nutanix rest api to check vm instance, pls review again, thanks.

**Pull request merged**